### ISSUE TITLE
Calculator doesn't read as well in NVDA 2020.4 as in 2020.3

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#12107]
### Steps to reproduce:

1. Open Calculator.
2. Type 6
3. Type x
4. Type 1
5. Type 0
6. Press enter
7. Type /
8. Type 15

### Actual behavior:

At each step, NVDA 2020.4 reads:
1. Calculator
2. 6
3. Star
4. 1
5. 0
6. Display is 60
7. slash
8. one, five

At each step NVDA 2020.3 reads:
1. Calculator
2. Display is 6
3. Display is 6 times
4. Display is 1
5. Display is Ten
6. Display is 60
7. Display is 60 divided by
8. Display is fifteen

### Expected behavior:

I don't mind whether "Display is" gets read every step, though it's probably useful.  It is more user friendly to have the full number read out ("Fifteen" instead of "one" then "five") and also having symbols such as slash read as divided by which is how it is used in the calculator.

#### NVDA version:

NVDA 2020.4 Installed 

#### Windows version:

Windows 10 (64-bit) Version: 2009, Build: 19042 

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Yes, latest (as at 1/3/2021) alpha version: alpha-21802,5678c8be

**Comment by josephsl** [comment]
Hi,

The changed behavior is intentional: imagine speak typed characters is enabled. NVDA will then say:

1. 6
2. Display is 6
3. Star
4. Display is 6 *
5. 1
6. 0
7. Display is 60
8. Slash
9. 1
10. 5

It was pointed out in #9428 that speak typed characters enabled and having to hear "display is ..." is annoying. The culprit is UIA notification event, which is an improvement over dealing with name property change and live regions in older Calculator app releases (Windows 10 App Essentials add-on commit log is filled with my frustrations with Calculator app's UIA implementation).

One way to resolve this is tell NVDA to announce the expression when a non-numeric character is entered. This means we need to either define a typed character event or assign a script that will be bound to dozens of keys. But that's not really a solution because UIA notification event will be fired whenever you enter something, which is precisely the issue described in #9428. I may need to dig into activity Id, and if the same activity Id is used whenever expressions change, then there's really nothing we can do other than persuade Calculator team at Microsoft that this is a really bad user experience behavior that affects any screen reader listening to UIA notification event (NVDA and Narrator included).

Thanks.

**Comment by josephsl** [comment]
Hi,

And no, the same notification activity Id is used regardless of keys you type to enter expressions.

Thanks.

**Comment by dkager** [comment]
More worrying is that NVDA does not report anything in braille as you type. Could this be a consequence of the changed behavior?

**Comment by josephsl** [comment]
Hi, specifically, UIA notification event. I’ll think about tweaking this for braille users. Thanks.

**Comment by josephsl** [comment]
Hi,

2022 update: braille announcement issue should be fixed in recent NVDA releases.

For now I will close this issue - feel free to reopen if needed. Thanks.
